tests/lava/proc/event_data/event_data_loader/test_aedat_data_loader.py

```diff
@@ -0,0 +1,295 @@
+
+    def test_invalid_file_path_extension(self):
+        """Tests for invalid file extension"""
+        with(self.assertRaises(ValueError)):
+            # Test for .py
+            _ = AedatDataLoader(file_path="test_aedat_data_loader.py",
+                                shape_out=(43200,))
+
+    def test_invalid_file_path_missing_file(self):
+        """Tests for missing file"""
+        with(self.assertRaises(FileNotFoundError)):
+            _ = AedatDataLoader(file_path="missing_file.aedat4",
+                                shape_out=(43200,))
+
+    def test_invalid_shape_out_dimension(self):
+        """Test for an invalid shape given to the shape_out param"""
+        with(self.assertRaises(ValueError)):
+            _ = AedatDataLoader(file_path="../dvs_recording.aedat4",
+                                shape_out=(240, 180))
+
+    def test_invalid_shape_out_negative_integer(self):
+        """Tests for a negative width given"""
+        with(self.assertRaises(ValueError)):
+            _ = AedatDataLoader(file_path="../dvs_recording.aedat4",
+                                shape_out=(-43200,))
+
+    def test_invalid_shape_out_decimal(self):
+        """Tests for a decimal width given"""
+        with(self.assertRaises(ValueError)):
+            _ = AedatDataLoader(file_path="../dvs_recording.aedat4",
+                                shape_out=(43200.5,))
+
+
+class TestProcessModelAedatDataLoader(unittest.TestCase):
+    def test_init(self):
+        proc_params = {
+            "file_path": "../dvs_recording.aedat4",
+            "shape_out": (3000,),
+            "seed_sub_sampling": 0
+        }
+
+        pm = AedatDataLoaderPM(proc_params)
+
+        self.assertIsInstance(pm, AedatDataLoaderPM)
+        self.assertEqual(pm._shape_out, proc_params["shape_out"])
+        self.assertIsInstance(pm._file, AedatFile)
+        self.assertIsInstance(pm._stream,
+                              _AedatFileEventNumpyPacketIterator)
+        self.assertIsInstance(pm._frame_shape, tuple)
+        self.assertIsInstance(pm._random_rng, np.random.Generator)
+
+    def test_sub_sampling(self):
+        data_history = [
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [1, 1, 1, 1, 1, 1],
+            [0],
+            [1, 1, 1],
+            [1],
+            [1],
+            [1]
+        ]
+        indices_history = [
+            [1597, 2308, 2486, 2496, 2498, 1787, 2642, 2633, 2489,
+             2488, 1596, 1729, 1727, 2500, 1780],
+            [1600, 1732, 2297, 1388, 2290, 2305, 3704, 3519, 1911],
+            [7138, 2301, 2471, 1601, 2982, 1364, 1379, 1386, 1384,
+             2983, 1390, 2289, 1401, 1362, 2293],
+            [1910, 1382, 1909, 1562, 1606, 1381],
+            [464],
+            [2323, 1908, 1393],
+            [4062],
+            [1792],
+            [3889]
+        ]
```

Comment:
I'm assuming this comes from the DVS file we are using. Can we reduce this in size to make the test smaller and more to the point?

Reply:
Now only checking for a single timestep whether subsampling works

Reply:
That may not be enough then. The test should make sure that we can read events, from multiple time steps, with different polarities, at different positions/indices. The test data should cover all of those cases (and others, if I forgot something) but does not need more. In the previous version, it seemed like it contained a bunch more cases.

Reply:
Mmmh, then I guess a good middle ground could be to go up to timestep 5 (with this data). We tried to reduce the resolution of the aedat4 file, but it seems like trying to use the "crop scale" options on dv for resizing leads to weird behavior with the metadata of the file, like reducing resolution doesn't actually change the x and y indices in the expected way. Do you think it's worth trying to crop another part of the video where we may get multiple polarities, and varying event batch sizes in the first 2-3 events?